startstoptorq.sh

```diff
@@ -0,0 +1,275 @@
+  procs=$(getall)                                                                                   # get all process names from csv
+  avail=()
+  for i in $input;
+  do 
+    if [[ `echo "$procs" | grep -w "$i"` ]]; then                                                   # check input process is valid
+      avail+="$i "                                                                                  # get only valid processes
+    else 
+      echo `date '+%H:%M:%S'` "| $i failed - unavailable processname"
+    fi
+  done
+  procs=$avail                                                                                      # assign valid processes
+ }
+
+getprocs() {
+  if [ "$2" == "all" ]; then 
+    procs=$(getall)                                                                                 # get all processes
+  else
+    shift                                                                                           # ignore first argument
+    checkinput $@                                                                                   # check input process names
+  fi
+ }
+
+flag() {
+  count=0
+  for i in ${BASH_ARGV[*]};
+  do
+    count=$(($count+1))
+    if [[ $i == "-$1" ]]; then                                                                      # find flag argument
+      n=$(($count-2))
+    fi
+  done
+ }
+
+flagextras() {
+  flag $@
+  z=$n
+  while [ $z -ge 0 ]
+  do
+    extras+="${BASH_ARGV[$z]} "                                                                     # get all extra parameters 
+    z=$[$z-1]
+  done
+ }
+
+flagcsv() {
+  flag $@
+  csvpath="${BASH_ARGV[$n]}"                                                                        # assign specifed csv file
+ }
+
+getextras() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]]; then                                            
+    eval flagextras "extras";
+    length=$(($#-$n-2));
+    array=${@:1:$length};                                                                           # arguments without extras flag
+  else
+    array=$@;
+  fi
+ }
+
+getcsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
+    eval flagcsv "csv";
+    length=$(($#-2));
+    array=${@:1:$length};                                                                           # arguments without csv flag
+    getprocs $array;
+  else
+    csvpath=$DEFAULTCSV;                                                                            # set csv file to default
+    getprocs $array;
+  fi
+ }
+
+getextrascsv() {
+  eval flagextras "extras";
+  eval flagcsv "csv";
+  length=$(($#-$n-2));
+  array=${@:1:$length};                                                                             # arguments without extras and csv flag
+  getprocs $array;
+ }
+
+checkextrascsv() {
+  if [[ `echo ${BASH_ARGV[*]} | grep -e extras` ]] && [[ `echo ${BASH_ARGV[*]} | grep -e csv` ]]; then
```

Comment:
In some places single `[` are used in if statements, here it's `[[` - make this consistent please